platform: nrf9160: Provide SPM HAL, enable SECURE_UART1 option

anangl opened the pull request:
Provide required SPM HAL functions for the nRF9160 platform,
together with the target_cfg implementation (mostly by copying
from nRF5340).
Make UARTE1 a Secure only peripheral by taking advantage of
the SECURE_UART1 option.

Further two commits contain small corrections for nRF5340.

anangl committed:
platform: nrf5340: Add missing peripherals in spu_periph_init_cfg()

Add missing entries for SPLIT security peripherals (DPPIC and NVMC)
in the function providing the default configuration of peripheral
access permissions. Use NRF_REGULATORS and NRF_CLOCK instead of
NRF_OSCILLATORS and NRF_RESET (these peripherals have the same base
addresses, respectively) in that function, for consistency with what
is used for the nRF9160 platform. For the same reason, remove the
unnecessary entry for NRF_FPU.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
platform: nrf9160: Provide SPM HAL, enable SECURE_UART1 option

Provide required SPM HAL functions for the nRF9160 platform,
together with the target_cfg implementation (mostly by copying
from nRF5340).
Make UARTE1 a Secure only peripheral by taking advantage of
the SECURE_UART1 option.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
platform: nrf5340: Correct return value types of SPU config functions

Use `enum tfm_plat_err_t` instead of `int32_t` as this is what those
functions in fact return.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl commented on platform/ext/target/nordic_nrf/nrf9160/target_cfg.c:
```diff
@@ -0,0 +1,243 @@
+
+enum tfm_plat_err_t init_debug(void)
+{
+    return TFM_PLAT_ERR_SUCCESS;
```
Should there be a compilation error issued here if an option different than `DAUTH_CHIP_DEFAULT` is used? Something like what is used in LPC55S69: https://github.com/NordicPlayground/trusted-firmware-m/blob/778b9b141a6cde2fd31c7e7825efabdcde7503f2/platform/ext/target/nxp/lpcxpresso55s69/target_cfg.c#L128-L130.

ioannisg replied:
yes, I think so @anangl

anangl replied:
Added an error message.

anangl commented on platform/ext/target/nordic_nrf/nrf9160/spm_hal.c:
```diff
@@ -0,0 +1,323 @@
+/*
```
What about making this file a common one for nRF5340 and nRF9160? Currently, for both platforms it is identical.

oyvindronningstad replied:
I think it's good idea.

ioannisg replied:
I agree, as long as we are confident we are not going to have #ifdef PLATFORM directives there.

anangl replied:
Moved the file to `nordic_nrf/common/`.

anangl committed:
platform: nordic_nrf: Use common spm_hal.c for nRF5340 and nRF9160

Implementation of this file is identical for both nRF platforms,
so instead of maintaining two copies of it, use a common one located
in the nordic_nrf/common/ directory.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

oyvindronningstad reviewed (approved):
LGTM

ioannisg commented on platform/ext/target/nordic_nrf/nrf5340/target_cfg.c:
```diff
@@ -209,8 +209,8 @@ enum tfm_plat_err_t spu_periph_init_cfg(void)
-	spu_peripheral_config_non_secure((uint32_t)NRF_OSCILLATORS, false);
-	spu_peripheral_config_non_secure((uint32_t)NRF_RESET, false);
+	spu_peripheral_config_non_secure((uint32_t)NRF_REGULATORS, false);
+	spu_peripheral_config_non_secure((uint32_t)NRF_CLOCK, false);
```
OK @anangl so as far as I understand from the commit message, this is not wrong but just done for uniformity with nrf9160

anangl replied:
Yes, exactly, only for uniformity.

anangl commented:
Removed unnecessary entries that configured non-secure access to the FPU, on both nRF platforms.
